### DISCUSSION TITLE
How to solve ?  Image is working with GeometryDrawing, but not with Source (Bitmap)

### DISCUSSION THREAD

**Discussion by iksi4prs** [AvaloniaUI/Avalonia#12720] [General]
Hi,

The code uses a Grid which contains servral images, where only of of them is displayed based on a rule (EnumToBooleanConverter)
The images which are implemented using GeometryDrawing, work find and are displayed
But images which uses Source to bind to a png/bmp is not displayed

Code that works
```
       <Image Classes="iconImage" Margin="10,3,3,3" VerticalAlignment="Center"
               IsVisible="{Binding Type, Converter={StaticResource EnumToBooleanConverter}, 
              ConverterParameter={x:Static enums:FileContentType.Audio}}">
            <Image.Source>
                <DrawingImage>
                    <DrawingImage.Drawing>
                        <GeometryDrawing Brush="Green"
                                         Geometry="M224 1......"/>
                    </DrawingImage.Drawing>
                </DrawingImage>
            </Image.Source>
        </Image>
```

code that doesnt work
```
    <Image Classes="iconImage" Margin="10,3,3,3"
            VerticalAlignment="Stretch" HorizontalAlignment="Stretch"
            IsVisible="{Binding Type, Converter={StaticResource EnumToBooleanConverter}, 
           ConverterParameter={x:Static enums:FileContentType.Word}}"
           Source="{Binding ImageToView}">
    </Image>

```

selector used:
```
    <Style Selector="Image.iconImage">
        <Setter Property="Width" Value="20" />
        <Setter Property="Height" Value="20" />
    </Style>
```
in the model I have

```
    public ModelABC()
    {
        InitializeComponent();
        _imageToView = new Bitmap("C:/test333.bmp");
    }

    private Bitmap _imageToView;
    public Bitmap ImageToView
    {
        get
        {
            return _imageToView;
        }
    }
```

I see the file is loaded in the ctor,  but binding 'get' of 'ImageToView' is **not** called.

I've also tried using binding converter, as mentioned here:

https://docs.avaloniaui.net/docs/controls/image

But still, I dont see that the 'get' of binding property is called.

Any help is appreciated.
Si.

**Comment by timunie** [comment]
Have you enabled CompiledBindings? If not, enable them in order to make sure DataContext is correct. 

You can also use DevTools [F12] to check DataContext

**Reply by iksi4prs** [reply]
@timunie , problem solved.
I'm converting and old project to new Avalonia version.
I also see that you added AvaloniaUseCompiledBindingsByDefault, so will using it.
Thanks for your help !
